proposals/kroxylicious-operator-api-alternative.md

```diff
@@ -0,0 +1,654 @@
+            memory: "64Mi"
+            cpu: "250m"
+#  # Optional: default ordered list of filters to be used by a virtualcluster that doesn't apply it own.  (this doesn't fit well with the Infra role - drop it?).
+#  defaultFilterRefs:
+#  - group: filter.kroxylicious.io
+#    kind: Filter
+#    name: encryption```
+
+status:
+   # status of the proxy tbd
+   conditions:
+   - status: x
+     type: y
+     reason: r
+   # per listener status 
+   ingresses:
+   - name: myclusterip
+     condidtions:
+     - ...
+```
+
+### ProxyIngress
+
+The ProxyIngress CR is the responsibilty of the the Infrastructure admin.
+
+It declares a named ingress mechanism - in other words, a way for traffic to get to a proxy.  ProxyIngress knows about four types:
+
+* clusterIP - for on-cluster traffic realised using Kubernetes ClusterIP Service. Supports TCP or TLS.
+* loadBalancer - for off-cluster realised using a Kubernetes LoadBalancer Service.  Supports TLS only.
+* openShiftRoute - for off-cluster realised using OpenShift Routes (OpenShift specific)
+* gateway - for off-cluster realised using Gateway API.  Supports `TLSRoutes` with listener type of `passthrough`.
+
+```yaml
+apiVersion: proxy.kroxylicious.io/v1
+kind: ProxyIngress
+metadata:
+  name: myclusterip
+spec:
+  # reference to the proxy that will use this ingress.
+  proxyRef:
+    kind: Proxy  # if present must be Gateway, otherwise defaulted
+    group: proxy.kroxylicious.io # if present must be proxy.kroxylicious, otherwise defaulted=
+    name: myproxy  # name of proxy
+
+  # oneOf: clusterIP, loadBalancer, openShiftRoute, gateway
+  clusterIP:
+    protocol: TCP|TLS (ClusterIP supports both)
+    port: # optional, defaults to 9082
+  loadBalancer:
+    bootstrapAddressPattern: $(vitrtualClusterName).kafka.com 
+    brokerAddressPattern: broker-$(nodeId).$(vitrtualClusterName).kafka.com
+    port: # optional, defaults to 9082
+  openShiftRoute:
+    # ingress controller name (optional, if omit defaults to default  in the openshift-ingress-operator namespace).
+    resourceRef:
+      kind: IngressController  # if present must be IngressController, otherwise defaulted to IngressController
+      group: operator.openshift.io # if present must be operator.openshift.io, otherwise defaulted to operator.openshift.io
+      name: myingresscontroller
+      namespace: openshift-ingress-operator # namespace of the gateway, if omitted assumes namespace of this resource
+  gateway:
+    type: TLSRoute
+    resourceRef:
+      kind: Gateway  # if present must be Gateway, otherwise defaulted to Gateway
+      group: gateway.networking.k8s.io # if present must be gateway.networking.k8s.io, otherwise defaulted to gateway.networking.k8s.io
+      name: foo-gateway  # name of gateway resource to be used
+      namespace: namespace # namespace of the gateway, if omitted assumes namespace of this resource
+      sectionName: mytlspassthrough # refers to a TLS passthrough listener defined in the Gateway
```

Comment:
Can you elaborate for each of these mechanisms what resources are expected to already exist/are the responsibility of the user, vs what resources will be managed by the operator?

Reply:
I have talked about that in worked examples.  In each case, I stated what resources the operator would create.